probe-rs/src/bin/probe-rs/cmd/cargo_embed/config/mod.rs

```diff
@@ -77,12 +77,13 @@ pub struct General {
 }
 
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+// Note: default values are defined in `RttChannelConfig`.
 pub struct UpChannelConfig {
     pub channel: usize,
     #[serde(default)]
     pub mode: Option<ChannelMode>,
     #[serde(default)]
-    pub format: DataFormat,
+    pub format: Option<DataFormat>,
```

Comment:
This now defers to `RttChannelConfig`'s default, as opposed to `DataFormat`'s. This is not strictly necessary as they are the same, but this change means `format` is no longer an exception to how this stuct's fields are handled.

Reply:
I don't think Option is the right choice here to be honest. I don't understand the necessity of this.

Reply:
This is an opinionated change, as value-vise everything ends up being identical right now.

Previously, the logic was this: If something optional is not specified in `UpChannelConfig`, we fall back to the default in `RttChannelConfig` OR use `DataFormat::default()` and ignore what `RttChannelConfig::default().data_format` is.

The logic behind this change is to remove this exception, and to have all of the default values in one place (`RttChannelConfig::default()`). The logic goes like this: if something optional is not specified in `UpChannelConfig`, we fall back to the default in `RttChannelConfig` for that field.